src/main/java/org/sagebionetworks/web/client/widget/entity/download/UploadDialogWidgetV2.java

```diff
@@ -61,19 +69,41 @@ public void show() {
     renderComponent(true);
   }
 
+  private void handleDrop(FileList fileList) {
+    if (this.ref != null && this.ref.current != null) {
+      this.ref.current.handleUploads(fileList);
+      // Show the uploader
+      this.show();
+    } else {
+      DomGlobal.console.error(
+        "EntityUploadHandle ref is null, aborting handleDrop"
+      );
```

Comment:
This should never happen, and user can recover by just opening the uploader, so I decided not to spend time on more sophisticated error handling.